File: db/java/app/src/main/java/com/example/sales/dto/UpdateProductRequest.java
package com.example.sales.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.Size;
import lombok.Data;

/**
 * 商品更新リクエストDTO
 * すべてのフィールドはオプション
 */
@Data
public class UpdateProductRequest {

    @Size(max = 40, message = "商品正式名は40文字以内で入力してください")
    private String fullName;

    @Size(max = 10, message = "商品名は10文字以内で入力してください")
    private String name;

    @Size(max = 20, message = "商品カナ名は20文字以内で入力してください")
    private String kanaName;

    @Min(value = 0, message = "販売単価は0以上で入力してください")
    private Integer unitPrice;

    @Min(value = 0, message = "売上原価は0以上で入力してください")
    private Integer primeCost;

    @Size(max = 8, message = "仕入先コードは8文字以内で入力してください")
    private String supplierCode;
}

File: db/csharp/SalesManagement.Api/Dtos/UpdateProductRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SalesManagement.Api.Dtos;

/// <summary>
/// 商品更新リクエストDTO
/// すべてのフィールドはオプション
/// </summary>
public class UpdateProductRequest
{
    [StringLength(40, ErrorMessage = "商品正式名は40文字以内で入力してください")]
    public string? FullName { get; set; }

    [StringLength(10, ErrorMessage = "商品名は10文字以内で入力してください")]
    public string? Name { get; set; }

    [StringLength(20, ErrorMessage = "商品カナ名は20文字以内で入力してください")]
    public string? KanaName { get; set; }

    [Range(0, int.MaxValue, ErrorMessage = "販売単価は0以上で入力してください")]
    public int? UnitPrice { get; set; }

    [Range(0, int.MaxValue, ErrorMessage = "売上原価は0以上で入力してください")]
    public int? PrimeCost { get; set; }

    [StringLength(8, ErrorMessage = "仕入先コードは8文字以内で入力してください")]
    public string? SupplierCode { get; set; }
}
